src/body/body.rs

```diff
@@ -545,14 +559,23 @@ impl Sender {
         futures_util::future::poll_fn(|cx| self.poll_ready(cx)).await
     }
 
-    /// Send data on this channel when it is ready.
+    /// Send data on data channel when it is ready.
     pub async fn send_data(&mut self, chunk: Bytes) -> crate::Result<()> {
         self.ready().await?;
-        self.tx
+        self.data_tx
             .try_send(Ok(chunk))
             .map_err(|_| crate::Error::new_closed())
     }
 
+    /// Send trailers on trailers channel.
+    pub fn send_trailers(&mut self, trailers: HeaderMap) -> crate::Result<()> {
```

Comment:
Even though not specifically required due to the internal implementation, I'm thinking this should be an `async fn`, so that we could adjust the implementation later if need be. But I'm open to hearing why it shouldn't.